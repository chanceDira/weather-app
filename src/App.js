import './App.css';
import { useState } from 'react';
import logo from './assets/logo.svg'

const api = {
  key: "d91e5af2517572deed4392e96dd7c09a",
  base: "https://api.openweathermap.org/data/2.5/"
}

function App() {

  const [search, setSearch] = useState('')
  const [weather, setWeather] = useState({})

  const handleGetWeatherData = async (e) => {
    e.preventDefault()
    await fetch(`${api.base}weather?q=${search}&units=metric&APPID=${api.key}`)
    .then(res => res.json())
    .then(result => {
      setWeather(result)
    })
    setSearch("")
  }


  return (
    <div className=" min-h-screen flex text-blue-600 justify-center items-center">
      <div>
        <div>
        <div className=' flex justify-center font-bold text-4xl mb-10'>
            Weather App
          </div>
          <div>
            <img src={logo} alt='svg' />
          </div>
         
        </div>
        <div>
          <form className='flex flex-row justify-center mt-6'>
            <div  className='mr-2'>
          <input 
            type='text' 
            placeholder='Enter city/town...' 
            value={search}
            className='border p-2 border-blue-200 rounded-md text-blue-500 outline-none'
            onChange={(e) => setSearch(e.target.value)}
            />
            </div>
            <div className='ml-2 bg-blue-600 text-white flex justify-center items-center rounded-md py-2 px-8'>
              <button data-testid="search" onClick={handleGetWeatherData}>Search</button>
            </div>
          </form>
        </div>

        <div className=' bg-blue-600 text-white p-8 flex flex-col justify-center items-center rounded-md mt-6'>
          {typeof weather.main != "undefined" ? 
          <>
             <div>
          <p className=' text-xl font-bold mb-4 underline text-blue-100'>{weather?.name}</p>
        </div>

        <div>
          <p>{weather?.main?.temp}Â°C</p>
        </div>

        <div className=' flex flex-col justify-center items-center'>
              <p>{weather?.weather[0]?.main}</p>
          <p>{weather?.weather[0]?.description}</p>
        </div>
          </>
          : 'No fetched data yet !!'}
        </div>

       

      </div>
    </div>
  );
}

export default App;
