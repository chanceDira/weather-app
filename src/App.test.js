import { render, screen } from '@testing-library/react';
import App from './App';

describe(App, () => {

  it('renders weather title', () => {
    const app = render(<App />);
    expect(app).toMatchSnapshot()
    
    const linkElement = screen.getByText(/Weather App/i);
    expect(linkElement).toBeInTheDocument();
  });

  it('render search button', () => {
    const { getByTestId } = render(<App />)
    const searchBtn = getByTestId('search').textContent
    expect(searchBtn).toBe('Search')

  })
})
