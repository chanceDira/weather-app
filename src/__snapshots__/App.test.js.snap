// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders weather title 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=" min-h-screen flex text-blue-600 justify-center items-center"
      >
        <div>
          <div>
            <div
              class=" flex justify-center font-bold text-4xl mb-10"
            >
              Weather App
            </div>
            <div>
              <img
                alt="svg"
                src="logo.svg"
              />
            </div>
          </div>
          <div>
            <form
              class="flex flex-row justify-center mt-6"
            >
              <div
                class="mr-2"
              >
                <input
                  class="border p-2 border-blue-200 rounded-md text-blue-500 outline-none"
                  placeholder="Enter city/town..."
                  type="text"
                  value=""
                />
              </div>
              <div
                class="ml-2 bg-blue-600 text-white flex justify-center items-center rounded-md py-2 px-8"
              >
                <button
                  data-testid="search"
                >
                  Search
                </button>
              </div>
            </form>
          </div>
          <div
            class=" bg-blue-600 text-white p-8 flex flex-col justify-center items-center rounded-md mt-6"
          >
            No fetched data yet !!
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" min-h-screen flex text-blue-600 justify-center items-center"
    >
      <div>
        <div>
          <div
            class=" flex justify-center font-bold text-4xl mb-10"
          >
            Weather App
          </div>
          <div>
            <img
              alt="svg"
              src="logo.svg"
            />
          </div>
        </div>
        <div>
          <form
            class="flex flex-row justify-center mt-6"
          >
            <div
              class="mr-2"
            >
              <input
                class="border p-2 border-blue-200 rounded-md text-blue-500 outline-none"
                placeholder="Enter city/town..."
                type="text"
                value=""
              />
            </div>
            <div
              class="ml-2 bg-blue-600 text-white flex justify-center items-center rounded-md py-2 px-8"
            >
              <button
                data-testid="search"
              >
                Search
              </button>
            </div>
          </form>
        </div>
        <div
          class=" bg-blue-600 text-white p-8 flex flex-col justify-center items-center rounded-md mt-6"
        >
          No fetched data yet !!
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
